packages/framework/src/Component/ClassExtension/ClassExtensionRegistry.php

```diff
@@ -51,37 +54,29 @@ public function addExtendedService(string $parentClassName, string $childClassNa
      */
     protected function getOtherClassesExtensionMap(): array
     {
-        $finder = Finder::create()
-            ->files()
-            ->ignoreUnreadableDirs()
-            ->in($this->frameworkRootDir . '/src')
-            ->name('/.*(Data|Controller)\.php/');
-
         $otherClassesMap = [];
 
-        /** @var \Symfony\Component\Finder\SplFileInfo $file */
-        foreach ($finder as $file) {
-            $frameworkClassFqcn = $this->getFqcn($file->getPathname());
-            $projectClassFqcn = str_replace('Shopsys\FrameworkBundle', 'App', $frameworkClassFqcn);
+        foreach ($this->packagesRegistry as $package) {
+            $finder = Finder::create()
+                ->files()
+                ->ignoreUnreadableDirs()
+                ->in($package['path'] . '/src')
+                ->name('/.*\.php/');
 
-            if (class_exists($projectClassFqcn)) {
-                $otherClassesMap[$frameworkClassFqcn] = $projectClassFqcn;
-            }
-        }
 
-        $finder = Finder::create()
-            ->files()
-            ->ignoreUnreadableDirs()
-            ->in($this->frameworkRootDir . '/../frontend-api/src')
-            ->name('/.*\.php/');
+            /** @var \Symfony\Component\Finder\SplFileInfo $file */
+            foreach ($finder as $file) {
+                try {
+                    $packageClassFqcn = $this->getFqcn($file->getPathname());
+                } catch (ErrorException) {
+                    continue;
```

Comment:
maybe it is worth outputing the error so we are aware of potential bugs?

Reply:
The error here would say, that array index 0 does not exists.

Reply:
ok :shrug: